### PR TITLE
[Feature#3234][cluster]enhanced load balancing

### LINKED ISSUE #3234: [Feature][cluster] enhanced load balancing
**Describe the feature**
In the production environment, our machine configurations may not be exactly the same. Therefore, in the process of load balancing, we hope to distribute traffic according to the weight of the configured machines.

### PR DESCRIPTION
**Pull request by CalvinKirs**
weight-based load balancing algorithm
this closes #3234

## What is the purpose of the pull request

weight-based load balancing algorithm

Implementation of load balancing based on weight.
Implementation of load balancing based on workgroups.

Related mail：https://lists.apache.org/thread.html/r06af73b0897d8a0c97a16ca1b9316d5d0bc29de833f60c10ca46415a%40%3Cdev.dolphinscheduler.apache.org%3E
https://lists.apache.org/thread.html/r5a70a5f43aa49973923ef23d5b079b883cda746a14e91f9c083a985a%40%3Cdev.dolphinscheduler.apache.org%3E
this closes #3234

## Verify this pull request

This pull request is already covered by existing tests, such as *(please describe tests)*.

This change added tests and can be verified as follows:

### CODE REVIEW

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/Constants.java [hunk 1]
```diff
@@ -249,6 +249,8 @@ private Constants() {
 
     public static final String WORKER_RESERVED_MEMORY = "worker.reserved.memory";
 
+    public static final String WORKER_WEIGHT = "worker.weight";
```
Comment (yangyichao-mango) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/Constants.java:252]: Hi,

This variable is not used in this project, please delete.
If you have any question or suggestion, please put forward~

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/registry/WorkerRegistry.java [hunk 2]
```diff
@@ -152,6 +152,7 @@ public void unRegistry() {
             // trim and lower case is need
             workerZkPathBuilder.append(workGroup.trim().toLowerCase()).append(SLASH);
             workerZkPathBuilder.append(address);
+            workerZkPathBuilder.append(weight).append(SLASH);
```
Comment (yangyichao-mango) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/registry/WorkerRegistry.java:155]: Hi,

If user choose to use `RoundRobinSelector`, there will be no need to add the `weight` to zk heartbeat info. Maybe we can design this feature better.

Comment (CalvinKirs) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/registry/WorkerRegistry.java:155]: Hi, in my opinion, every load choice is related to weight. Can you talk about why it is not needed when using the RoundRobin algorithm?

**Comment by CalvinKirs** [comment]
todo:RoundRobinSelector

**Comment by CalvinKirs** [comment]
done

**Comment by CalvinKirs** [comment]
TBR.

**Pull request merged**